lib/chat_list/chat_list_screen.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_chat/chat_list/chat_list.dart';
+import 'package:flutter_chat/chat_list/chat_list_item.dart';
+import 'package:provider/provider.dart';
+
+class ChatListPage extends StatefulWidget {
+  @override
+  _ChatListPageState createState() => _ChatListPageState();
+}
+
+class _ChatListPageState extends State<ChatListPage>
+    with SingleTickerProviderStateMixin {
+  TabController _tabController;
+
+  @override
+  void initState() {
+    super.initState();
+    _tabController = TabController(length: 3, vsync: this);
+  }
+
+  @override
+  void dispose() {
+    _tabController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ChangeNotifierProvider(
+      create: (ctx) => ChatListSource(),
+      child: Scaffold(
+        appBar: AppBar(
+          title: Text("Chat list"),
+          bottom: TabBar(
+            controller: _tabController,
+            tabs: <Widget>[
+              Tab(
+                icon: Icon(Icons.chat),
+                text: "chat",
+              ),
+              Tab(
+                icon: Icon(Icons.group),
+                text: "group",
+              ),
+              Tab(
+                icon: Icon(Icons.settings),
+                text: "settings",
+              )
+            ],
+          ),
+        ),
+        body: TabBarView(
+          children: <Widget>[ChatListView(), ChatListView(), ChatListView()],
+          controller: _tabController,
+        ),
+      ),
+    );
+  }
+}
+
+class ChatListView extends StatefulWidget {
```

Comment:
It would be better to use Consumer here instead of Stateful Widget

Reply:
Changed to Consumer.